picasso/src/main/java/com/squareup/picasso/OkHttpDownloader.java

```diff
@@ -89,19 +90,29 @@ protected final OkHttpClient getClient() {
     return client;
   }
 
-  @Override public Response load(Uri uri, boolean localCacheOnly) throws IOException {
-    com.squareup.okhttp.Request.Builder requestBuilder =
-        new com.squareup.okhttp.Request.Builder().url(uri.toString());
-
-    if (localCacheOnly) {
-      requestBuilder.addHeader("Cache-Control", "only-if-cached,max-age=" + Integer.MAX_VALUE);
+  @Override public Response load(Uri uri, int networkPolicy) throws IOException {
+    CacheControl cacheControl;
+    if (NetworkPolicy.isOfflineOnly(networkPolicy)) {
+      cacheControl = CacheControl.FORCE_CACHE;
+    } else {
+      CacheControl.Builder builder = new CacheControl.Builder();
+      if (!NetworkPolicy.shouldReadFromDiskCache(networkPolicy)) {
+        builder.noCache();
+      }
+      if (!NetworkPolicy.shouldWriteToDiskCache(networkPolicy)) {
+        builder.noStore();
+      }
+      cacheControl = builder.build();
     }
 
+    com.squareup.okhttp.Request.Builder requestBuilder =
+        new com.squareup.okhttp.Request.Builder().cacheControl(cacheControl).url(uri.toString());
```

Comment:
Actually I am keeping the builder and adding a `networkPolicy > 0` conditional to avoid allocating an extra cache control object.